### PR TITLE
Apply camera name in topics

### PR DESCRIPTION
**Pull request by hyunseok-yang**
Camera name is not applied into topic name.

[Tracked by RSDEV-472]

### CODE REVIEW

**Comment by hyunseok-yang** [comment]
To achevie what I expected, it needs to be modified as latest commit.

@SamerKhshiboun  
As we disccused at this PR https://github.com/IntelRealSense/realsense-ros/pull/2764, the code should be modified.

the command
```shell
ros2 launch realsense2_camera rs_launch.py align_depth.enable:=true rgb_camera.profile:=640x480x6 depth_module.profile:=640x480x6 enable_sync:=True camera_name:=cam1 camera_namespace:=ns1
```

latest of ros2-development
```
/ns1/aligned_depth_to_color/camera_info
/ns1/aligned_depth_to_color/image_raw
/ns1/aligned_depth_to_color/image_raw/compressed
/ns1/aligned_depth_to_color/image_raw/compressedDepth
/ns1/aligned_depth_to_color/image_raw/theora
/ns1/color/camera_info
/ns1/color/image_raw
/ns1/color/image_raw/compressed
/ns1/color/image_raw/compressedDepth
/ns1/color/image_raw/theora
/ns1/color/metadata
/ns1/depth/camera_info
/ns1/depth/image_rect_raw
/ns1/depth/image_rect_raw/compressed
/ns1/depth/image_rect_raw/compressedDepth
/ns1/depth/image_rect_raw/theora
/ns1/depth/metadata
/ns1/extrinsics/depth_to_color
/ns1/extrinsics/depth_to_depth
/ns1/imu
```


after apply the latest commit
```
/ns1/cam1/aligned_depth_to_color/camera_info
/ns1/cam1/aligned_depth_to_color/image_raw
/ns1/cam1/aligned_depth_to_color/image_raw/compressed
/ns1/cam1/aligned_depth_to_color/image_raw/compressedDepth
/ns1/cam1/aligned_depth_to_color/image_raw/theora
/ns1/cam1/color/camera_info
/ns1/cam1/color/image_raw
/ns1/cam1/color/image_raw/compressed
/ns1/cam1/color/image_raw/compressedDepth
/ns1/cam1/color/image_raw/theora
/ns1/cam1/color/metadata
/ns1/cam1/depth/camera_info
/ns1/cam1/depth/image_rect_raw
/ns1/cam1/depth/image_rect_raw/compressed
/ns1/cam1/depth/image_rect_raw/compressedDepth
/ns1/cam1/depth/image_rect_raw/theora
/ns1/cam1/depth/metadata
/ns1/cam1/extrinsics/depth_to_color
/ns1/cam1/extrinsics/depth_to_depth
/ns1/cam1/imu
```

**Comment by SamerKhshiboun** [comment]
@hyunseok-yang Thanks for submitting the PR.
I see that a lot of test failed due to this change, and that what is blocking the PR right now.
Do you think you can solve these test issues ?

Assigning @Arun-Prasad-V  as reviewer.

**Comment by SamerKhshiboun** [comment]
Tracked by RSDEV-472 internally.

We should also update the README and announce this change to all users.

**Comment by hyunseok-yang** [comment]
@SamerKhshiboun 
Well, it looks test suite was failed since I changed the topic of publishers.

I'm not sure I can modify all of test scirpt in the workspace.


And, for the README file, where should I modify it?

**Comment by Arun-Prasad-V** [comment]
Hi @hyunseok-yang ,

Please take a look at this https://design.ros2.org/articles/topic_and_service_names.html.
 
image

Refer the last row: if you just prefix "~/" to the input topic name, it will automatically add the node namespace and the node name.

Can we try this?

**Comment by hyunseok-yang** [comment]
> Hi @hyunseok-yang ,
> 
> Please take a look at this https://design.ros2.org/articles/topic_and_service_names.html.
> 
> image
> Refer the last row: if you just prefix "~/" to the input topic name, it will automatically add the node namespace and the node name.
> 
> Can we try this?

Thank you for your suggestion.
It seems work!

I've updated the commit.

File: realsense2_camera/src/rs_node_setup.cpp [hunk 1]
```diff
@@ -327,7 +327,7 @@ void BaseRealSenseNode::startRGBDPublisherIfNeeded()
 }
```
Comment (Arun-Prasad-V) [inline realsense2_camera/src/rs_node_setup.cpp:327]: @hyunseok-yang ,  Please update the same for "rgbd" topic as well. Refer line number 319.

**Comment by Arun-Prasad-V** [comment]
@hyunseok-yang ,

Please update the poindcloud topic name as well. Refer file "named_filter.cpp" line number 125. [https://github.com/IntelRealSense/realsense-ros/blob/72b552497892992fb31a8661f186192dd4d1c15c/realsense2_camera/src/named_filter.cpp#L125](url)

File: realsense2_camera/src/rs_node_setup.cpp (hunk 1 above)
Comment (hyunseok-yang) [reply realsense2_camera/src/rs_node_setup.cpp:327]: Oh, I missed it!

**Comment by hyunseok-yang** [comment]
> @hyunseok-yang ,
> 
> Please update the poindcloud topic name as well. Refer file "named_filter.cpp" line number 125. [https://github.com/IntelRealSense/realsense-ros/blob/72b552497892992fb31a8661f186192dd4d1c15c/realsense2_camera/src/named_filter.cpp#L125](url)

@Arun-Prasad-V 
Updated that part as well.
Please review the latest commit :)

**Comment by hyunseok-yang** [comment]
@Arun-Prasad-V 

I think test suite scripts should be modified also in spite of topic is modified as we expected.

Could you give any guidance or hint where should I edit it?

**Comment by Arun-Prasad-V** [comment]
> @Arun-Prasad-V
> 
> I think test suite scripts should be modified also in spite of topic is modified as we expected.
> 
> Could you give any guidance or hint where should I edit it?

@hyunseok-yang ,
Sure. We will review and let you know about test suite updates and readme update.
Thanks.

cc: @PrasRsRos

**Review by SamerKhshiboun — APPROVED** [review]

**Comment by hyunseok-yang** [comment]
@SamerKhshiboun 
how about test suite??

**Comment by SamerKhshiboun** [comment]
> @SamerKhshiboun how about test suite??

Tests have been fixed and merged by @PrasRsRos.
Thanks for your contributions

**Pull request merged**